In [1]:
import synapseclient
from synapseclient import Project, File, Folder, Wiki
from synapseclient import Schema, Column, Table, Row, RowSet, as_table_columns
import itertools
import pandas as pd

In [2]:
syn = synapseclient.Synapse()
syn.login()

Welcome, Victor Baham!



### Create a Citations wiki subpage in the ADKP backend portal for each study where its name is the study name in the `Study` column of [Portal - Studies Table](https://www.synapse.org/#!Synapse:syn17083367/tables/)

#### Note: start at index 4 where index i = 0, 1, 2… n because I already manually created Citations wiki subpages for the first 4 studies

In [4]:
query_ps = syn.tableQuery(f"SELECT * FROM {syn.get('syn17083367').id}")
portal_studies = pd.read_csv(query_ps.filepath)

#### Note: The name of each Synapse entity in the `Study` column can be accessed with `syn.get('synID').name`

In [5]:
portal_studies

ROW_ID  ROW_VERSION  Study_Type                 Program        Study  \
0        60          679  Consortium              ["AMP-AD"]   syn5550419   
1        61          679  Individual              ["AMP-AD"]  syn16779040   
2        62          679  Individual              ["AMP-AD"]   syn5759376   
3        63          679  Individual  ["AMP-AD", "M20VE-AD"]   syn9705278   
4        64          680  Individual              ["AMP-AD"]   syn5592519   
..      ...          ...         ...                     ...          ...   
126     272          669  Consortium            ["MODEL-AD"]  syn50944316   
127     273          670  Consortium                     NaN  syn27207345   
128     274          672  Individual            ["Psych-AD"]  syn52160016   
129     275          675  Individual            ["M2OVE-AD"]  syn31511672   
130     276          677  Consortium            ["MODEL-AD"]  syn27207345   

                                          DataType_All  \
0                                  ["Gene Expression"]   
1                                  ["Gene Expression"]   
2                                       ["Proteomics"]   
3                                     ["Metabolomics"]   
4                                     ["Metabolomics"]   
..                                                 ...   
126  ["Gene Expression, Electrophysiology, Immunoas...   
127                                                NaN   
128                                ["Gene Expression"]   
129                                ["Gene Expression"]   
130                   ["Gene Expression, Immunoassay"]   

                                            Study_Name  \
0      The RNAseq Sample Swap Study (rnaSeqSampleSwap)   
1         The AD Cross Species Study (AD_CrossSpecies)   
2             The Adult Changes in Thought (ACT) Study   
3    The Alzheimer's Disease Metabolomics Consortiu...   
4    The Alzheimer's Disease Metabolomics Consortiu...   
..                                                 ...   
126                            The UCI_Bin1K358R Study   
127                                The UCI_ABCA7 Study   
128  The Mount Sinai Neuropsychiatric Symptoms in A...   
129  The Mayo Clinic Single Nucleus RNAseq Study (M...   
130                                The UCI_ABCA7 Study   

                                     Study_Description specimenType  \
0    This study provides transcriptomic data from a...    ["Brain"]   
1    This study provides transcriptomic data from a...    ["Brain"]   
2    This study provides proteomic data from the  d...    ["Brain"]   
3    This study provides metabolomics data generate...    ["Blood"]   
4    This study provides metabolomics data generate...    ["Blood"]   
..                                                 ...          ...   
126  This study provides bulk RNA-seq, immunohistoc...    ["Brain"]   
127                                                NaN    ["Brain"]   
128  This study provides single-nucleus RNAseq prof...    ["Brain"]   
129  This study provides single-nucleus transcripto...    ["Brain"]   
130                                                NaN          NaN   

                                            studyFocus  ... isModelSystem  \
0                          ["No Cognitive Impairment"]  ...         False   
1                                ["Alzheimer Disease"]  ...          True   
2                                ["Alzheimer Disease"]  ...         False   
3                                ["Alzheimer Disease"]  ...         False   
4                                ["Alzheimer Disease"]  ...         False   
..                                                 ...  ...           ...   
126  ["Alzheimer Disease", " Electrophysiology", "I...  ...          True   
127                              ["Alzheimer Disease"]  ...          True   
128  ["Alzheimer Disease", "Parkinson Disease", "Fr...  ...         False   
129                              ["Alzheimer Disease"]  ...    

#### * Starting at index 5: `Citations` value is `syn12666371/wiki/628264` (index 4 actually starts the pattern but was already filled in manually)

#### * Ending at index 130: `Citations` value is `syn12666371/wiki/628389`

In [6]:
citation_wiki_ids_from_5 = ['syn12666371/wiki/' + str(x) for x in range(628264, 628390)]

In [7]:
citation_wiki_ids_from_5

['syn12666371/wiki/628264',
 'syn12666371/wiki/628265',
 'syn12666371/wiki/628266',
 'syn12666371/wiki/628267',
 'syn12666371/wiki/628268',
 'syn12666371/wiki/628269',
 'syn12666371/wiki/628270',
 'syn12666371/wiki/628271',
 'syn12666371/wiki/628272',
 'syn12666371/wiki/628273',
 'syn12666371/wiki/628274',
 'syn12666371/wiki/628275',
 'syn12666371/wiki/628276',
 'syn12666371/wiki/628277',
 'syn12666371/wiki/628278',
 'syn12666371/wiki/628279',
 'syn12666371/wiki/628280',
 'syn12666371/wiki/628281',
 'syn12666371/wiki/628282',
 'syn12666371/wiki/628283',
 'syn12666371/wiki/628284',
 'syn12666371/wiki/628285',
 'syn12666371/wiki/628286',
 'syn12666371/wiki/628287',
 'syn12666371/wiki/628288',
 'syn12666371/wiki/628289',
 'syn12666371/wiki/628290',
 'syn12666371/wiki/628291',
 'syn12666371/wiki/628292',
 'syn12666371/wiki/628293',
 'syn12666371/wiki/628294',
 'syn12666371/wiki/628295',
 'syn12666371/wiki/628296',
 'syn12666371/wiki/628297',
 'syn12666371/wiki/628298',
 'syn12666371/wiki/6

In [9]:
portal_studies['Citations'].iloc[range(5,131)] = citation_wiki_ids_from_5

[WARNING] /var/folders/lg/w3yckgy936lc0r6yw790hjs00000gr/T/ipykernel_18852/2721657272.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  portal_studies['Citations'].iloc[range(5,131)] = citation_wiki_ids_from_5



In [10]:
portal_studies

ROW_ID  ROW_VERSION  Study_Type                 Program        Study  \
0        60          679  Consortium              ["AMP-AD"]   syn5550419   
1        61          679  Individual              ["AMP-AD"]  syn16779040   
2        62          679  Individual              ["AMP-AD"]   syn5759376   
3        63          679  Individual  ["AMP-AD", "M20VE-AD"]   syn9705278   
4        64          680  Individual              ["AMP-AD"]   syn5592519   
..      ...          ...         ...                     ...          ...   
126     272          669  Consortium            ["MODEL-AD"]  syn50944316   
127     273          670  Consortium                     NaN  syn27207345   
128     274          672  Individual            ["Psych-AD"]  syn52160016   
129     275          675  Individual            ["M2OVE-AD"]  syn31511672   
130     276          677  Consortium            ["MODEL-AD"]  syn27207345   

                                          DataType_All  \
0                                  ["Gene Expression"]   
1                                  ["Gene Expression"]   
2                                       ["Proteomics"]   
3                                     ["Metabolomics"]   
4                                     ["Metabolomics"]   
..                                                 ...   
126  ["Gene Expression, Electrophysiology, Immunoas...   
127                                                NaN   
128                                ["Gene Expression"]   
129                                ["Gene Expression"]   
130                   ["Gene Expression, Immunoassay"]   

                                            Study_Name  \
0      The RNAseq Sample Swap Study (rnaSeqSampleSwap)   
1         The AD Cross Species Study (AD_CrossSpecies)   
2             The Adult Changes in Thought (ACT) Study   
3    The Alzheimer's Disease Metabolomics Consortiu...   
4    The Alzheimer's Disease Metabolomics Consortiu...   
..                                                 ...   
126                            The UCI_Bin1K358R Study   
127                                The UCI_ABCA7 Study   
128  The Mount Sinai Neuropsychiatric Symptoms in A...   
129  The Mayo Clinic Single Nucleus RNAseq Study (M...   
130                                The UCI_ABCA7 Study   

                                     Study_Description specimenType  \
0    This study provides transcriptomic data from a...    ["Brain"]   
1    This study provides transcriptomic data from a...    ["Brain"]   
2    This study provides proteomic data from the  d...    ["Brain"]   
3    This study provides metabolomics data generate...    ["Blood"]   
4    This study provides metabolomics data generate...    ["Blood"]   
..                                                 ...          ...   
126  This study provides bulk RNA-seq, immunohistoc...    ["Brain"]   
127                                                NaN    ["Brain"]   
128  This study provides single-nucleus RNAseq prof...    ["Brain"]   
129  This study provides single-nucleus transcripto...    ["Brain"]   
130                                                NaN          NaN   

                                            studyFocus  ... isModelSystem  \
0                          ["No Cognitive Impairment"]  ...         False   
1                                ["Alzheimer Disease"]  ...          True   
2                                ["Alzheimer Disease"]  ...         False   
3                                ["Alzheimer Disease"]  ...         False   
4                                ["Alzheimer Disease"]  ...         False   
..                                                 ...  ...           ...   
126  ["Alzheimer Disease", " Electrophysiology", "I...  ...          True   
127                              ["Alzheimer Disease"]  ...          True   
128  ["Alzheimer Disease", "Parkinson Disease", "Fr...  ...         False   
129                              ["Alzheimer Disease"]  ...    

In [12]:
syn.store(Table('syn17083367', portal_studies))

Uploading: 100%|█████████████████| 81.2k/81.2k [00:00<00:00, 129kB/s, table.csv]
